mhs/common/mhs_common/workflow/tests/test_synchronous.py

```diff
@@ -0,0 +1,236 @@
+    @async_test
+    async def test_send_message_success_message(self, wd_mock):
+        self._setup_success_workflow()
+        wdo = mock.MagicMock()
+        wdo.publish.return_value = test_utilities.awaitable(None)
+        wd_mock.return_value = wdo
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(None)
+
+        error, text, work_description_response = await self.wf.handle_outbound_message(from_asid="202020",
+                                                                                       message_id="123",
+                                                                                       correlation_id="qwe",
+                                                                                       interaction_details={},
+                                                                                       payload="nice message",
+                                                                                       work_description_object=None)
+
+        wdo.set_outbound_status.assert_called_with(work_description.MessageStatus.OUTBOUND_MESSAGE_RESPONSE_RECEIVED)
+        self.assertEqual(error, 200)
+        self.assertEqual(text, 'response body')
+
+    @mock.patch('mhs_common.state.work_description.create_new_work_description')
+    @async_test
+    async def test_handle_outbound_errors_when_no_asid_provided(self, wd_mock):
+        wdo = mock.MagicMock()
+        wdo.publish.return_value = test_utilities.awaitable(None)
+        wd_mock.return_value = wdo
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(None)
+
+        error, text, work_description_response = await self.wf.handle_outbound_message(from_asid=None,
+                                                                                       message_id="123",
+                                                                                       correlation_id="qwe",
+                                                                                       interaction_details={},
+                                                                                       payload="nice message",
+                                                                                       work_description_object=None)
+
+        self.assertEqual(error, 400)
+        self.assertEqual(text, '`from_asid` header field required for sync messages')
+
+
+    @async_test
+    async def test_no_inbound(self):
+        with self.assertRaises(NotImplementedError):
+            await self.wf.handle_inbound_message("1", "2", mock.MagicMock(), "payload")
+
+    @async_test
+    async def test_prepare_message(self):
+        id, headers, message = await self.wf._prepare_outbound_message("message_id", "to_asid", "from_asid", "mesasge",
+                                                                       {'service': 'service', 'action': 'action'})
+        self.assertEqual(id, "message_id")
```

Comment:
Again, probably worth checking the other properties of the message are set as expected too.